protocol/test-serialization/src/lib.rs

```diff
@@ -500,6 +501,22 @@ mod tests {
         test_roundtrip(&block.value, block.bytes.as_slice());
     }
 
+    #[test]
+    #[wasm_bindgen_test]
+    fn test_corrupted_deserialization() {
+        let mut rng = rand::thread_rng();
```

Comment:
Awsome work using rand generates random data for more robust test 👍 !

Reply:
Looks great

Reply:
I think you probably need to create individual unit tests for each sub-type of `ExternalTransition` to assure each of those individually fails on corrupt data.

Strictly speaking, this test does not prove that *some* corruption to these members won't go undetected.  As an extreme example, it could be the case that this top-level `ExternalTransition` negative test only passes because corruption of a single members is being detected (while the rest are not).   That could mean that another test case with corruption *only in those other members* could still fail.  Thus, I think you ought to create a corruption test for each sub-type.

Reply:
Added coverage for more compound structs. Please note that some simple structs can be deserialized into with no error as some of the checks will happen later when the bin-prot Serializable struct is converted into another equivalent struct for calculations, e.g. it does not ensure random public keys can be converted into BigInteger256 successfully at this stage, the check will happen upon the conversion, which is pending mina crypto refactoring

Reply:
👍🏼